### DISCUSSION TITLE
Cannot build on Windows 10 enterprise. Flutter env.g.dart file not generated?

### DISCUSSION THREAD

**Discussion by dearliuliu0522** [KRTirtho/spotube#1520] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I want to build on my Windows 10 enterprise laptop，but get something wrong. Here is build error log:

[SEVERE] envied_generator:envied on lib/collections/env.dart:

**Environment variable file doesn't exist at `.env`.
package:spotube/collections/env.dart:7:16
  ╷
7 │ abstract class Env {
  │                ^^^
  ╵**

### Expected Behavior

I don't know if Spotube cann't be build on Windows 10, or I do something wrong?  If it's the later ,what should I do?

### Steps to reproduce

**1. My environment is below**
PS D:\1_project\flutter\spotube> flutter doctor 
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.22.0, on Microsoft Windows [版本 10.0.19045.4412], locale zh-CN)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[√] Chrome - develop for the web
[√] Android Studio (version 2023.3)
[√] IntelliJ IDEA Ultimate Edition (version 2021.3)
[√] Connected device (4 available)
[√] Network resources

• No issues found!

PS C:\Users\admin> dart --version
Dart SDK version: 3.4.0 (stable) (Mon May 6 07:59:58 2024 -0700) on "windows_x64"

2. build log(do as [document](https://github.com/KRTirtho/spotube/blob/master/CONTRIBUTION.md#your-first-code-contribution) tells except first step cause I use Windows 10).

When I try to build Spotube, I get an error at the end. I upgrade some packages like envied etc. (suggested in https://stackoverflow.com/questions/76820607/flutter-env-g-dart-file-not-generated) but still not solve the problem. Here is my build log:

PS D:\1_project\flutter\spotube> **flutter pub get**
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
Resolving dependencies... (11.5s)
Downloading packages...
+ _fe_analyzer_shared 67.0.0 (68.0.0 available)
+ analyzer 6.4.1 (6.5.0 available)
+ analyzer_plugin 0.11.3
+ ansicolor 2.0.2
+ app_links 6.1.1
+ app_package_maker 0.0.9 (0.3.6 available)
+ app_package_maker_aab 0.0.9 (0.2.3 available)
+ app_package_maker_apk 0.0.9 (0.2.3 available)
+ app_package_maker_deb 0.0.9 (0.2.3 available)
+ app_package_maker_dmg 0.0.9 (0.2.3 available)
+ app_package_maker_exe 0.0.9 (0.2.3 available)
+ app_package_maker_ipa 0.0.9 (0.2.3 available)
+ app_package_maker_zip 0.0.9 (0.2.3 available)
+ archive 3.6.0
+ args 2.5.0
+ async 2.11.0
+ audio_service 0.18.13
+ audio_service_mpris 0.1.3
+ audio_service_platform_interface 0.1.1
+ audio_service_web 0.1.2
+ audio_session 0.1.19
+ auto_size_text 3.0.0
+ bonsoir 5.1.9
+ bonsoir_android 5.1.4
+ bonsoir_darwin 5.1.2
+ bonsoir_linux 5.1.2
+ bonsoir_platform_interface 5.1.2
+ bonsoir_windows 5.1.4
+ boolean_selector 2.1.1
+ build 2.4.1
+ build_cli_annotations 2.1.0
+ build_config 1.1.1
+ build_daemon 4.0.2
+ build_resolvers 2.4.2
+ build_runner 2.4.10
+ build_runner_core 7.3.0
+ built_collection 5.1.1
+ built_value 8.9.2
+ buttons_tabbar 1.3.10
+ cached_network_image 3.3.1
+ cached_network_image_platform_interface 4.0.0
+ cached_network_image_web 1.2.0
+ catcher_2 1.2.6
+ change_case 1.1.0 (2.0.1 available)
+ characters 1.3.0
+ checked_yaml 2.0.3
+ ci 0.1.0
+ cli_util 0.4.1
+ clock 1.1.1
+ code_builder 4.10.0
+ collection 1.18.0
+ color 3.0.0
+ convert 3.1.1
+ cross_file 0.3.4+1
+ crypto 3.0.3
+ csslib 1.0.0
+ cupertino_icons 1.0.6
+ curved_navigation_bar 1.0.3
+ custom_lint 0.6.4
+ custom_lint_builder 0.6.4
+ custom_lint_core 0.6.3
+ dart_des 1.0.2
+ dart_discord_rpc 0.0.3 from git https://github.com/Tommypop2/dart_discord_rpc.git at 4d0501
+ dart_mappable 4.2.2
+ dart_style 2.3.6
+ dartx 1.2.0
+ dbus 0.7.10
+ device_frame 1.1.0
+ device_info_plus 10.1.0
+ device_info_plus_platform_interface 7.0.0
+ device_preview 1.1.0
+ dio 5.4.3+1
+ disable_battery_optimization 1.1.1
+ dots_indicator 2.1.2 (3.0.0 available)
+ draggable_scrollbar 0.1.0 from git https://github.com/thielepaul/flutter-draggable-scrollbar.git at cfd570
+ duration 3.0.13
+ envied 0.5.4+1
+ envied_generator 0.5.4+1
+ equatable 2.0.5
+ fake_async 1.3.1
+ ffi 2.1.2
+ file 7.0.0
+ file_picker 8.0.3
+ file_selector 1.0.3
+ file_selector_android 0.5.1+2
+ file_selector_ios 0.5.3
+ file_selector_linux 0.9.2+1
+ file_selector_macos 0.9.4
+ file_selector_platform_interface 2.6.2
+ file_selector_web 0.9.4+1
+ file_selector_windows 0.9.3+1
+ fixnum 1.1.0
+ fluentui_system_icons 1.1.239
+ flutter 0.0.0 from sdk flutter
+ flutter_app_builder 0.0.9 (0.4.2 available)
+ flutter_app_packager 0.0.9 (0.4.2 available)
+ flutter_broadcasts 0.4.0
+ flutter_cache_manager 3.3.1 (3.3.2 available)
+ flutter_desktop_tools 0.0.1 from git https://github.com/KRTirtho/flutter_desktop_tools.git at 1f0bec
+ flutter_displaymode 0.6.0
+ flutter_distributor 0.0.2 (0.4.3 available)
+ flutter_driver 0.0.0 from sdk flutter
+ flutter_feather_icons 2.0.0+1
+ flutter_gen_core 5.5.0+1
+ flutter_gen_runner 5.5.0+1
+ flutter_hooks 0.20.5
+ flutter_inappwebview 6.0.0
+ flutter_inappwebview_android 1.0.13
+ flutter_inappwebview_internal_annotations 1.1.1
+ flutter_inappwebview_ios 1.0.13
+ flutter_inappwebview_macos 1.0.11
+ flutter_inappwebview_platform_interface 1.0.10
+ flutter_inappwebview_web 1.0.8
+ flutter_keyboard_visibility 6.0.0
+ flutter_keyboard_visibility_linux 1.0.0
+ flutter_keyboard_visibility_macos 1.0.0
+ flutter_keyboard_visibility_platform_interface 2.0.0
+ flutter_keyboard_visibility_web 2.0.0
+ flutter_keyboard_visibility_windows 1.0.0
+ flutter_launcher_icons 0.13.1
+ flutter_lints 4.0.0
+ flutter_localizations 0.0.0 from sdk flutter
+ flutter_mailer 2.1.2
+ flutter_native_splash 2.4.0
+ flutter_plugin_android_lifecycle 2.0.18
+ flutter_riverpod 2.5.1
+ flutter_rust_bridge 1.82.1 (1.82.6 available)
+ flutter_secure_storage 9.2.2
+ flutter_secure_storage_linux 1.2.1
+ flutter_secure_storage_macos 3.1.2
+ flutter_secure_storage_platform_interface 1.1.2
+ flutter_secure_storage_web 1.2.1
+ flutter_secure_storage_windows 3.1.2
+ flutter_sharing_intent 1.1.1
+ flutter_svg 1.1.6 (2.0.10+1 available)
+ flutter_test 0.0.0 from sdk flutter
+ flutter_web_plugins 0.0.0 from sdk flutter
+ fluttertoast 8.2.5
+ form_validator 2.1.1
+ freezed 2.5.2 (2.5.3 available)
+ freezed_annotation 2.4.1
+ frontend_server_client 4.0.0
+ fuchsia_remote_debug_protocol 0.0.0 from sdk flutter
+ fuzzywuzzy 1.1.6
+ gap 3.0.1
+ glob 2.1.2
+ go_router 14.1.3
+ google_fonts 6.2.1
+ graphs 2.3.1
+ gsettings 0.2.8
+ gtk 2.1.0
+ hashcodes 2.0.0
+ hive 2.2.3
+ hive_flutter 1.1.0
+ hive_generator 2.0.1
+ hooks_riverpod 2.5.1
+ hotreloader 4.2.0
+ html 0.15.4
+ html_unescape 2.0.0
+ http 1.2.1
+ http_methods 1.1.1
+ http_multi_server 3.2.1
+ http_parser 4.0.2
+ image 4.2.0
+ image_picker 1.1.1
+ image_picker_android 0.8.12+1
+ image_picker_for_web 3.0.3
+ image_picker_ios 0.8.11+2
+ image_picker_linux 0.2.1+1
+ image_picker_macos 0.2.1+1
+ image_picker_platform_interface 2.10.0
+ image_picker_windows 0.2.1+1
+ image_size_getter 2.1.3
+ integration_test 0.0.0 from sdk flutter
! intl 0.19.0 (overridden)
+ introduction_screen 3.1.14
+ io 1.0.4
+ jiosaavn 0.1.0
+ js 0.6.7 (0.7.1 available)
+ json_annotation 4.9.0
+ json_serializable 6.8.0
+ leak_tracker 10.0.4 (10.0.5 available)
+ leak_tracker_flutter_testing 3.0.3 (3.0.5 available)
+ leak_tracker_testing 3.0.1
+ lints 4.0.0
+ local_notifier 0.1.6
+ logger 2.3.0
+ logging 1.2.0
+ lrc 1.0.2
+ mailer 6.1.0
+ matcher 0.12.16+1
+ material_color_utilities 0.8.0 (0.11.1 available)
+ media_kit 1.1.10+1
+ media_kit_libs_android_audio 1.3.6
+ media_kit_libs_audio 1.0.4
+ media_kit_libs_ios_audio 1.1.4
+ media_kit_libs_linux 1.1.3
+ media_kit_libs_macos_audio 1.1.4
+ media_kit_libs_windows_audio 1.0.9
+ media_kit_native_event_loop 1.0.8
+ meta 1.12.0 (1.15.0 available)
+ metadata_god 0.5.2+1
+ mime 1.0.5
+ nested 1.0.0
+ oauth2 2.0.2
+ octo_image 2.0.0
+ package_config 2.1.0
+ package_info_plus 8.0.0
+ package_info_plus_platform_interface 3.0.0
+ palette_generator 0.3.3+3
+ path 1.9.0
+ path_drawing 1.0.1
+ path_parsing 1.0.1
+ path_provider 2.1.3
+ path_provider_android 2.2.3
+ path_provider_foundation 2.4.0
+ path_provider_linux 2.2.1
+ path_provider_platform_interface 2.1.2
+ path_provider_windows 2.2.1
+ permission_handler 11.3.1
+ permission_handler_android 12.0.6
+ permission_handler_apple 9.4.4
+ permission_handler_html 0.1.1
+ permission_handler_platform_interface 4.2.1
+ permission_handler_windows 0.2.1
+ petitparser 6.0.2
+ piped_client 0.1.1 from git https://github.com/KRTirtho/piped_client.git at 646317
+ platform 3.1.4
+ plugin_platform_interface 2.1.8
+ pool 1.5.1
+ popover 0.3.0+1
+ process 5.0.2
+ provider 6.1.2
+ pub_api_client 2.7.0
+ pub_semver 2.1.4
+ pubspec 2.3.0
+ pubspec_parse 1.2.3
+ puppeteer 3.11.0
+ quiver 3.2.1
+ recase 4.1.0
+ riverpod 2.5.1
+ riverpod_analyzer_utils 0.5.1 (0.5.3 available)
+ riverpod_lint 2.3.10 (2.3.12 available)
+ rxdart 0.27.7
+ safe_local_storage 1.0.2
+ screen_retriever 0.1.9
+ scrobblenaut 3.0.0 from git https://github.com/KRTirtho/scrobblenaut.git at d90cb7
+ scroll_to_index 3.0.1
+ sentry 8.2.0
+ shared_preferences 2.2.3
+ shared_preferences_android 2.2.1
+ shared_preferences_foundation 2.4.0
+ shared_preferences_linux 2.3.2
+ shared_preferences_platform_interface 2.3.2
+ shared_preferences_web 2.3.0
+ shared_preferences_windows 2.3.2
+ shelf 1.4.1
+ shelf_router 1.1.4
+ shelf_static 1.1.2
+ shelf_web_socket 1.0.4 (2.0.0 available)
+ sidebarx 0.17.1
+ simple_icons 10.1.3
+ skeleton_text 3.0.1
+ skeletonizer 1.1.2+1
+ sky_engine 0.0.99 from sdk flutter
+ sliver_tools 0.2.12
+ smtc_windows 0.1.2
+ source_gen 1.5.0
+ source_helper 1.3.4
+ source_span 1.10.0
+ spotify 0.13.6+1
+ sqflite 2.3.3+1
+ sqflite_common 2.5.4
+ stack_trace 1.11.1
+ state_notifier 1.0.0
+ stream_channel 2.1.2
+ stream_transform 2.1.0
+ string_scanner 1.2.0
+ stroke_text 0.0.3
+ sync_http 0.3.1
+ synchronized 3.1.0+1
+ system_theme 2.3.1
+ system_theme_web 0.0.2
! system_tray 2.0.2 (overridden) (2.0.3 available)
+ term_glyph 1.2.1
+ test_api 0.7.0 (0.7.1 available)
+ time 2.1.4
+ timezone 0.9.3
+ timing 1.0.1
+ titlebar_buttons 1.0.0
+ tuple 2.0.2
+ type_plus 2.1.1
+ typed_data 1.3.2
+ universal_io 2.2.2
+ universal_platform 1.1.0
+ uri 1.0.0
+ uri_parser 2.0.2
+ url_launcher 6.2.6
+ url_launcher_android 6.3.2
+ url_launcher_ios 6.3.0
+ url_launcher_linux 3.1.1
+ url_launcher_macos 3.2.0
+ url_launcher_platform_interface 2.3.2
+ url_launcher_web 2.3.1
+ url_launcher_windows 3.1.1
+ uuid 3.0.7 (4.4.0 available)
+ vector_graphics_codec 1.1.11+1
+ vector_graphics_compiler 1.1.11+1
+ vector_math 2.1.4
+ version 3.0.2
+ very_good_infinite_list 0.7.1
+ visibility_detector 0.4.0+2
+ vm_service 14.2.1 (14.2.2 available)
+ watcher 1.1.0
+ web 0.5.1
+ web_socket_channel 2.4.5 (3.0.0 available)
+ webdriver 3.0.3
+ wikipedia_api 0.1.0
+ win32 5.5.1
+ win32_registry 1.1.2
+ window_manager 0.3.9
+ window_size 0.1.0 from git https://github.com/google/flutter-desktop-embedding.git at a73891 in plugins/window_size
+ xdg_directories 1.0.4
+ xml 6.5.0
+ yaml 3.1.2
+ youtube_explode_dart 2.2.1
Changed 324 dependencies!
32 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
PS D:\1_project\flutter\spotube> **dart run build_runner build --delete-conflicting-outputs --enable-experiment=records,patterns**
Resolving dependencies in `D:\1_project\flutter\spotube`... (1.0s)
Downloading packages... (1.3s)
Got dependencies in `D:\1_project\flutter\spotube`.
Building package executable... (6.2s)
Built build_runner:build_runner.
[INFO] Generating build script completed, took 622ms
[INFO] Precompiling build script... completed, took 6.5s
FlutterGen v5.5.0+1 Loading ... pubspec.yaml
[INFO] Building new asset graph completed, took 2.2s
[INFO] Checking for unexpected pre-existing outputs. completed, took 4ms
[INFO] Generating SDK summary completed, took 6.3s
[WARNING] json_serializable on lib/models/connect/connect.dart:
The version constraint "^4.8.1" on json_annotation allows versions before 4.9.0 which is not allowed.
[SEVERE] envied_generator:envied on lib/collections/env.dart:

**Environment variable file doesn't exist at `.env`.
package:spotube/collections/env.dart:7:16
  ╷
7 │ abstract class Env {
  │                ^^^
  ╵**
[INFO] 54.3s elapsed, 1607/1608 actions completed.Generated: lib\collections\assets.gen.dart
FlutterGen finished.
[INFO] Running build completed, took 55.4s
[INFO] Caching finalized dependency graph completed, took 199ms
[SEVERE] Failed after 55.6s

### Operating System

Window 10 Enterprise 22H2

### Spotube version

v3.6.0

### Installation source

GitHub Releases (Binary)

### Self grab

- [ ] I'm ready to work on this issue!

**Comment by KRTirtho** [comment]
Please first read the [Build from source guide](https://github.com/KRTirtho/spotube/blob/master/CONTRIBUTION.md#your-first-code-contribution)

You didn't create the `.env` file with required credentials. Make sure to create the `.env` and then run `build_runner build`

**Comment by aqz236** [comment]
I encountered the same problem in Windows 11, but it has been solved. You can try adding configuration for flutter_gen in `pubspec.yaml`:
```yaml
flutter_gen:
   integrations:
     flutter_localizations: true
   output: lib/collections
```
